### PR TITLE
CMake: fix gdal.pc and gdal-config generation when CMAKE_INSTALL_xxxx is an absolute path (fixes #6004)

### LINKED ISSUE #6004: `gdal.pc` contains wrong paths if `CMAKE_INSTALL_<dir>` is absolute
## Expected behavior and actual behavior.
I'm building gdal on NixOS and obtaining the following in `lib/pkgconfig/gdal.pc`:
```
CONFIG_VERSION=3.5.0
CONFIG_INST_PREFIX=/nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3
CONFIG_INST_LIBS=-L/nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3//nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3/lib -lgdal
CONFIG_INST_CFLAGS=-I/nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3//nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3/include
CONFIG_INST_DATA=/nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3/share/gdal
prefix=${CONFIG_INST_PREFIX}
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${exec_prefix}/include
datadir=${CONFIG_INST_DATA}

Name: libgdal
Description: Geospatial Data Abstraction Library
Version: ${CONFIG_VERSION}
Libs: ${CONFIG_INST_LIBS}
Libs.private: -lcryptopp -lssl -lcrypto -llzma -ldeflate -llz4 -larmadillo -lqhull_r -lgeotiff -ltiff -lheif -lLerc -lzstd -lbrunslienc-c -lbrunslidec-c -ljpeg -lpoppler -lpng -lgif -lnetcdf -lblosc -lcfitsio -lhdf5 -lcurl -lwebp -lhdf -lmfhdf -lz -lopenjp2 -L/nix/store/iyf9g7w9r448wimavwwbdjw320wlrjc4-tiledb-2.3.3/lib64 -ltiledb -lIlmImf -lIlmImfUtil -lHalf -lIex -lpcre2-8 -lspatialite -lsqlite3 -lxerces-c -L/nix/store/fmbxbsil9y1dvgl0ph7llirdz7mx29f0-mariadb-connector-c-3.2.5/lib/mysql -lmysqlclient -lpq -lparquet -larrow -ldl -lexpat -lxml2 -lgeos_c -lproj -ljson-c -lstdc++ -lm
Cflags: ${CONFIG_INST_CFLAGS}
```
I would expect
```
CONFIG_INST_LIBS=-L/nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3/lib -lgdal
CONFIG_INST_CFLAGS=-I/nix/store/crsca8wbc9c0g3kmjcpc6b7pgxxygkz7-gdal-3.5.0.3/include
```
See also https://github.com/jtojnar/cmake-snips#assuming-cmake_install_dir-is-relative-path.
This was not an issue when using autotools with GDAL 3.4.

## Steps to reproduce the problem.
Run `nix-build -A gdal` on https://github.com/NixOS/nixpkgs/pull/179406.

## Operating system
NixOS 22.11-pre

## GDAL version and provenance
3.5.0.3 from GitHub tarball

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: gdal.cmake [hunk 1]
```diff
@@ -573,6 +573,7 @@ add_subdirectory(man)
 
 # GDAL 4.0 ? Install headers in ${CMAKE_INSTALL_INCLUDEDIR}/gdal ?
 set(GDAL_INSTALL_INCLUDEDIR ${CMAKE_INSTALL_INCLUDEDIR})
+set(GDAL_INSTALL_FULL_INCLUDEDIR ${CMAKE_INSTALL_FULL_INCLUDEDIR})
```
Comment (dg0yt) [inline gdal.cmake:576]: IMO this new variable should be introduced. Just use `CMAKE_INSTALL_FULL_INCLUDEDIR`...
... which is actually what you do.

File: cmake/helpers/GdalGenerateConfig.cmake [hunk 2]
```diff
@@ -265,14 +265,14 @@ function(gdal_generate_config)
         set(CONFIG_PREFIX "${CMAKE_INSTALL_PREFIX}")
     endif()
     set(CONFIG_DATA "${CONFIG_PREFIX}/${GDAL_RESOURCE_PATH}")
-    set(CONFIG_CFLAGS "-I${CONFIG_PREFIX}/${GDAL_INSTALL_INCLUDEDIR}")
+    set(CONFIG_CFLAGS "-I${GDAL_INSTALL_FULL_INCLUDEDIR}")
```
Comment (dg0yt) [inline cmake/helpers/GdalGenerateConfig.cmake:268]: IMO this should be switched to use `CMAKE_INSTALL_INCLUDEDIR` directly.

**Review by dg0yt — COMMENTED** [review]
Please reduce the levels of indirection in the GDALs build system.
And leave time for comments.

**Comment by dg0yt** [comment]
Uff, had to edit both comments. Really, more time for review desired ;-)

File: gdal.cmake (hunk 1 above)
Comment (rouault) [reply gdal.cmake:576]: the point of this new variable is related to the comment "# GDAL 4.0 ? Install headers in ${CMAKE_INSTALL_INCLUDEDIR}/gdal". The variable will make it easier to track affected locations if we ever change where we install GDAL headers

Comment (dg0yt) [reply gdal.cmake:576]: For now, I see premature pessimization/YAGNI. You could as well grep for the CMAKE variable if it is really going to be needed.
All these rarely used customization points are a barrier to sharing maintainer work and to onboarding new contributors.

Comment (rouault) [reply gdal.cmake:576]: fair point. I'll do a follow-up PR to simplify this

**Pull request merged**